docs/overrides.rst

```diff
@@ -0,0 +1,130 @@
+Overrides
+=========
+
+During an image life cycle there can be a need to do a slightly tweaked builds - using different base images, injecting newer libraries etc. We want to support such scenarios without a need of duplicating whole image sources. To achieve this Concreate supports overrides mechanism for its image descriptor. You can override almost anything in image descriptor. The overrides are based on overrides descriptor - a yaml file containing changes to image descriptor.
+
+To use an override descriptor you need to pass ``--overides <PATH>`` argument to a concreate.
+
+**Example**: To use overrides.yaml  file located in current working directory run:
+
+.. code:: bash
+
+	  $ concreate build --overrides=overrides.yaml
+
+How overrides works
+-------------------
+
+Concreate is using `YAML <http://yaml.org/>`_ format for its descriptors. Overrides in concreate works on `YAML node <http://www.yaml.org/spec/1.2/spec.html#id2764044>`_ level.
+
+
+Scalar nodes
+^^^^^^^^^^^^
+Scalar nodes are easy to overrides, if Concreate finds any scalar node in a overrides descriptor it update its value in image descriptor with the overridden one.
+
+**Example**: Overriding scalar node:
+
+*image descriptor*
+
+.. code:: yaml
+
+	  schema_version: 1
+	  name: "dummy/example"
+	  version: "0.1"
+	  from: "busybox:latest"
+
+*overrides descriptor*
+
+.. code:: yaml
+
+	  schema_version: 1
+	  from: "fedora:latest"
+
+*overridden image descriptor*
+
+.. code:: yaml
+
+	  schema_version: 1
+	  name: "dummy/example"
+	  version: "0.1"
+	  from: "fedora:latest"
+
+Sequence nodes
+^^^^^^^^^^^^^^
+Sequence nodes are little bit tricky, if they're representing plain arrays, we cannot easily override any value so Concreate is just merging arrays from image and override descriptors together. This can have unexpected results, please see `corresponding issue. <https://github.com/jboss-container-images/concreate/issues/106>`_
```

Comment:
I would probably add a top level section: known issues, or even create a separate page for known issues and link to it from here.

Reply:
I've highlighted it little - but i prefer to have it tight with the docs, so its easy to spot